pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -453,10 +470,15 @@ func (r *ReconcileClusterSync) Reconcile(ctx context.Context, request reconcile.
 		return reconcile.Result{}, err
 	}
 
-	needToDoFullReapply := needToCreateClusterSync || r.timeUntilFullReapply(lease) <= 0
-	if needToDoFullReapply {
-		logger.Info("need to reapply all syncsets")
+	needToRenew := r.timeUntilRenew(lease) <= 0
+
+	if needToCreateLease {
+		logger.Info("need to reapply all syncsets (missing lease)")
+	} else if needToRenew {
+		logger.Info("need to reapply all syncsets (time to renew)")
 	}
+
+	needToDoFullReapply := needToCreateLease || needToRenew
```

Comment:
Say, shouldn't this also include `|| needToCreateClusterSync`? (Which would need to be set above in the `IsNotFound` branch after Get(clusterSync) -- which was probably the original intent of the `needToCreateClusterSync` var.)

Reply:
If you check the original code, nothing was ever setting it to true

Reply:
> If you check the original code, nothing was ever setting it to true

I know -- it should have :) See

> (Which would need to be set above in the `IsNotFound` branch after Get(clusterSync) -- which was probably the original intent of the `needToCreateClusterSync` var.)

[Later] I walked through the code and figured out how deleting the clustersync is triggering a full reapply: it's because we're using the clustersync's status -- which is empty since it just got created -- to figure out which syncsets we need to apply. So whereas visually it would be arguably more readable if we had that folded in explicitly, it would indeed be redundant. Probably worth a comment though.